// Copyright 2017 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;
option java_outer_classname = "TabEventProtos";
option java_package = "org.chromium.components.metrics";

package metrics;

// Defines specialized fields used by TabEvent for UKMs.
message TabEvent {
  // Represents the MIME type of content.
  // MIME type strings have the format "<type>/<subtype>". The values for
  // ContentType reflect just the top-level type (e.g., AUDIO is used for both
  // "audio/mp3" and "audio/ogg").
  // The exception is "text/html", since it's treated quite differently than
  // other text.
  enum ContentType {
    // Could not be determined or not available.
    CONTENT_TYPE_UNKNOWN = 0;

    // Not one of the types below.
    CONTENT_TYPE_OTHER = 1;

    // The "application/*" type. Supported application sub-types are enumerated
    // in Chrome's PluginType.
    CONTENT_TYPE_APPLICATION = 2;

    // The "audio/*" type.
    CONTENT_TYPE_AUDIO = 3;

    // The "image/*" type.
    CONTENT_TYPE_IMAGE = 4;

    // The "text/*" type. For "text/html", TEXT_HTML is used instead.
    CONTENT_TYPE_TEXT = 5;

    // The "text/html" type, representing web pages.
    CONTENT_TYPE_TEXT_HTML = 6;

    // The "video/*" type.
    CONTENT_TYPE_VIDEO = 7;
  }

  optional ContentType content_type = 1;
}
